components/analytics/true_up_review.tsx

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {useDispatch, useSelector} from 'react-redux';
+
+import moment from 'moment';
+
+import {getTrueUpReviewBundle, getTrueUpReviewStatus} from 'actions/hosted_customer';
+import {
+    isCurrentLicenseCloud,
+} from 'mattermost-redux/selectors/entities/cloud';
+
+import {
+    getSelfHostedErrors,
+    getTrueUpReviewProfile as trueUpReviewProfileSelector,
+    getTrueUpReviewStatus as trueUpReviewStatusSelector,
+} from 'mattermost-redux/selectors/entities/hosted_customer';
+
+import useCanSelfHostedSignup from 'components/common/hooks/useCanSelfHostedSignup';
+import CheckMarkSvg from 'components/widgets/icons/check_mark_icon';
+
+import './true_up_review.scss';
+import {GlobalState} from '@mattermost/types/store';
+import WarningIcon from 'components/widgets/icons/fa_warning_icon';
+import {isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+
+const TrueUpReview: React.FC = () => {
+    const dispatch = useDispatch();
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const isAirGapped = !useCanSelfHostedSignup();
+    const reviewProfile = useSelector(trueUpReviewProfileSelector);
+    const reviewStatus = useSelector(trueUpReviewStatusSelector);
+    const isSystemAdmin = useSelector(isCurrentUserSystemAdmin);
+    const license = useSelector(getLicense);
+    const [submitted, setSubmitted] = useState(false);
+    const trueUpReviewError = useSelector((state: GlobalState) => {
+        const errors = getSelfHostedErrors(state);
+        return Boolean(errors.trueUpReview);
+    });
+
+    useEffect(() => {
+        dispatch(getTrueUpReviewStatus());
```

Comment:
Nit: 3/5 check the request's status because this would lead to a new request for each new render. I'm more fan of setting the request status in redux, as we have listeners for the request, error, and success. An example of this is this [reducer](https://github.com/mattermost/mattermost-webapp/blob/1607dcddafe53ad2be4836e859c04eb5f0cc2a66/packages/types/src/cloud.ts#L105), and then you can do this [check](https://github.com/mattermost/mattermost-webapp/blob/1607dcddafe53ad2be4836e859c04eb5f0cc2a66/components/announcement_bar/overage_users_banner/index.tsx#L74) to ensure we make one request.